DigitalOffice/Services/TimeManagementService/Validators/CreateWorkTimeRequestValidator.cs

```diff
@@ -0,0 +1,87 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using LT.DigitalOffice.TimeManagementService.Models;
+using LT.DigitalOffice.TimeManagementService.Repositories.Filters;
+using LT.DigitalOffice.TimeManagementService.Repositories.Interfaces;
+using System.Globalization;
+
+namespace LT.DigitalOffice.TimeManagementService.Validators
+{
+    public class CreateWorkTimeRequestValidator : AbstractValidator<CreateWorkTimeRequest>
+    {
+        private IWorkTimeRepository repository;
+
+        /// <summary>
+        /// How many days ago can WorkTime be added.
+        /// </summary>
+        public const int fromDay = 3;
+        /// <summary>
+        /// How many days ahead can WorkTime be added.
+        /// </summary>
+        public const int toDay = 2;
+        /// <summary>
+        /// Limit on working hours in a row.
+        /// </summary>
+        public readonly static TimeSpan workingLimit = new TimeSpan(24, 0, 0); // 24 hours.
+
+        private readonly DateTime fromDateTime = DateTime.Now.AddDays(-fromDay);
+        private readonly DateTime toDateTime = DateTime.Now.AddDays(toDay);
+        private readonly CultureInfo culture = CultureInfo.CreateSpecificCulture("en-GB");
+
+        public CreateWorkTimeRequestValidator([FromServices] IWorkTimeRepository repository)
+        {
+            this.repository = repository;
+
+            RuleFor(wt => wt.WorkerUserId)
+                    .NotEmpty();
+
+            RuleFor(wt => wt.StartTime)
+                .NotEqual(new DateTime())
+                .Must(st => st > fromDateTime)
+                .WithMessage(date => $"WorkTime had to be filled no later than {fromDateTime.ToString(culture)}.")
+                .Must(st => st < toDateTime)
+                .WithMessage(date => $"WorkTime cannot be filled until {toDateTime.ToString(culture)}.");
+
+            RuleFor(wt => wt.EndTime)
+                .NotEqual(new DateTime());
+
+            RuleFor(wt => wt.ProjectId)
+                .NotEmpty();
+
+            RuleFor(wt => wt.Title)
+                .NotEmpty();
+
+            RuleFor(wt => wt)
+                .Must(wt => wt.StartTime < wt.EndTime)
+                .WithMessage("You cannot indicate that you worked zero hours or a negative amount.")
+                .Must(wt => wt.StartTime - wt.EndTime <= workingLimit)
+                .WithMessage(time => string.Format(
+                    "You cannot indicate that you worked more than {0} hours and {1} minutes.",
+                    workingLimit.Hours, workingLimit.Minutes))
+                .Must(wt =>
+                {
+                    var oldWorkTimes = repository.GetUserWorkTimes(
+                        wt.WorkerUserId,
+                        new WorkTimeFilter
+                        {
+                            StartTime = wt.StartTime.AddMinutes(-workingLimit.TotalMinutes),
+                            EndTime = wt.EndTime.AddMinutes(workingLimit.TotalMinutes)
```

Comment:
Can't figure out why we search for the times 24 hours before and after the one we want to set. 
If we can add times 3 days before and 2 days after (or vice versa), then we should add to StartTime x3 and EndTime x2 modifier? What do you think, Pavel?

Reply:
Because this is a limit for any given time.

Reply:
Fixed.